site/src/app/course/[id]/page.tsx

```diff
@@ -1,5 +1,39 @@
 import CoursePage from '../CoursePage';
 
+export async function generateMetadata({ params }: { params: Promise<{ id: string }> }) {
+  const id = decodeURIComponent((await params).id);
+
+  const input = encodeURIComponent(JSON.stringify({ courseID: id }));
+
+  const url = `${process.env.BACKEND_ROOT_URL}/trpc/courses.get?input=${input}`;
+
+  const res = await fetch(url);
+  const json = await res.json();
+
+  const course = json.result.data;
```

Comment:
Look for how we fetch the current user (presumably in the root layout.tsx, but could be elsewhere) to see how we set up trpc routes when we need data from the server. I'd recommend a similar approach to that instead of manually crafting the result with `fetch()`